### ISSUE TITLE
Challenge Issue When Log In

### ISSUE THREAD

**Issue by EezeSolutions** [a-legotin/InstaSharper#112]
Instagram recently started showing challenge to enter 6 digit code. But i don't see how to get past in this api there is no related function.

Can you please update regarding this...

Thank you

**Comment by vitalragaz** [comment]
@EezeSolutions When I can reproduce it. I can fix it. Actually I've tried it with four accounts and a for loop that contains 15 iterations per account and never a got an error within my test.

**Comment by EezeSolutions** [comment]
If your account working on your home IP just grab any public IP an log in you will see the issue.

thank you

**Comment by Nyoibouu** [comment]
@EezeSolutions 

That's because instagram has flagged all the proxies that have been used for spamming purposes. There is no fix for this, it's the updated anti spam measure system.

**Comment by EezeSolutions** [comment]
@Nyoibouu  It says Challenge required just like it says on the web which is solved... So it does not say account blocked or cannot user etc.. It says challenge required so must be a way to solve the challenge..

Thank you

**Comment by Nyoibouu** [comment]
@EezeSolutions 

No

**Comment by vitalragaz** [comment]
@Nyoibouu That's not correct. Don't speak for the crowd and say there is no fix. 

This could be easily solved. 
-> Creating Interface for pop3
-> Adding Mail Credentials to Builder or extending credentials
-> Checking for authrequest respone for -> "message = challenge_required"
-> Grabbing url property from challenge.
-> Open Page over webrequest and send payload.
-> Pop3 check and grabbing sec code
-> Send payload to challenge url POST -> security_code={code}

It's almost the same spam protection as facebook have so the method above worked by facebook btw.

In my opinion it's more an ethic question.

**Comment by EezeSolutions** [comment]
@vitalragaz  

You are absolutely correct when do we that in terms of Web :). However on the api there is no Url returned ..

So "Grabbing url property from challenge." how to get this..

**Comment by vitalragaz** [comment]
@EezeSolutions Just looking at the json object and you have everything you need ;)

Btw. I hide the codes in the image.
<img width="1234" alt="bildschirmfoto 2017-12-11 um 12 07 28" src="https://user-images.githubusercontent.com/3964460/33828394-3dd57200-de6c-11e7-9a94-23cdc3cf850f.png">

**Comment by EezeSolutions** [comment]
Ah perfect thank you  💃

**Comment by SergeyGorbatyuk171** [comment]
@vitalragaz @EezeSolutions ,
I have faced the same problem. I tried grabbing the url out of JSON object and then just pass checkpoint using browser, but that doesn't help. I see there is a solution written a few hours ago, but reading this, i can't understand, what to do, i didn't face with that technologies and can't google anything relevant. Would I please ask you to write a bit more comprehensive and clear solution example(for noobs like me :) )? Or even it is worth to make a short article in wiki or README, because the problem appears to be very common.

**Comment by vitalragaz** [comment]
@SergeyGorbatyuk171 I've wrote down the full automatized procedure a few article above how it can be implemented. I've wrote a small prototype for proof of concept that worked (took about 30min or so). A full implementation wouldn't take longer than I think 60-90min with unit tests and everything else. But this feature is against my ethical attitude, so I wouldn't be the person who publish it to the repo - sorry guys. But I can just share my knowledge 🔢

**Comment by andriyshevchenko** [comment]
@vitalragaz I'd appreciate your work, you can create a separate branch for it. For me it's to hard to follow these steps, because, i'm a novice. Thanks.

**Comment by Nyoibouu** [comment]
@vitalragaz 

You are clueless. That's a simple fucking email authentication check. This new check is not related to the fucking email. 

https://github.com/mgp25/Instagram-API/issues/1734

Read and aware yourself.

**Comment by vitalragaz** [comment]
@Nyoibouu interesting information, thanks! 

But there is still a difference between spam flag and a public ip that was noted as grey. As example if I'm on a public hotspot that was marked and I use an account that was never in contact with an api or something else. You will receive the digit code that the thread starter mentioned and the login won't work and a personal user, who won't use it for spam purpose, isn't able to login with that api without passing the email check. 

So, it's kindly based on the purpose of the thread starter what he actually want to do with the api and based on that, there are two different systems. One can't be solved and the other is the email check that can be automated.

**Comment by joselitinho1** [comment]
@vitalragaz about your solution for pop3, I should have the email/pass from each account, alright? In this case, I would be viable for me, but I guess is a good fix for who have few accounts.

**Comment by joselitinho1** [comment]
@vitalragaz another question, did you find any pattern from uuid?

**Comment by gdupiton** [comment]
Why is this issue closed without a solution?

**Comment by a-legotin** [comment]
@gdupiton , solution is obvios: do not use this library for any kind of suspicious actions e.g massive bombing/botting/statistical info retrieva and do not use public proxies, of course. And you will not be blamed in unusial activity. As simple as it is!

**Comment by gdupiton** [comment]
@a-legotin, 
With all due respect, please do not dismiss my issue with false and offensive accusations.  I respect you and the work that you've done here.  Please return the same respect.  We are a community of developers and should support each other and should not point the finger without merit.  I HAVE NOT performed ANY activities that should be flagged.  This arose after login attempts during testing of my process.  No bombing, botting...etc. This is an issue that is happening to others.  THEREFORE, ARE WE ALL GUILTY OF SUSPICIOUS ACTIVITY!?!?!  Or should we be investigating the problem in a proactive fashion?

**Comment by a-legotin** [comment]
@gdupiton , wtf man, did you find my reply like offensive accusation? I just told you most reasanoble premise for you issue. If you will find any solution feel free to send pull request, I have no plans to investigate such problems, like: reason of sentry block/ban/challenge. This is waste of time, Instagram team changing their algorithms quite often. Anytime you can pass challenge with you mobile device and official app, not a big deal.

**Comment by joselitinho1** [comment]
@gdupiton send me your email or telegram

**Comment by gdupiton** [comment]
loudupiton@gmail.com

**Comment by gdupiton** [comment]
@joselitinho1 loudupiton@gmail.com

**Comment by vitalragaz** [comment]
@EezeSolutions it's already within that issue.

vitalragaz commented on 10 Dec 2017
> This could be easily solved.
> -> Creating Interface for pop3
> -> Adding Mail Credentials to Builder or extending credentials
> -> Checking for authrequest respone for -> "message = challenge_required"
> -> Grabbing url property from challenge.
> -> Open Page over webrequest and send payload.
> -> Pop3 check and grabbing sec code
> -> Send payload to challenge url POST -> security_code={code}

**Comment by EezeSolutions** [comment]
@vitalragaz 
Thank you for the reply actually i have done all the above and it works perfectly fine in case of web..
But still on the Api Login i am not sure how to do the step:

--> Checking for authrequest respone for -> "message = challenge_required"
-->  Grabbing url property from challenge.  (### THIS ONE###)

 var logInResult = _instaApi.LoginAsync();
                                                while (logInResult.Status == TaskStatus.WaitingForActivation)
                                                { }
                                                if (logInResult.Status == TaskStatus.RanToCompletion)
                                                {
                                                    
                                                    if (logInResult.Result.Succeeded)
                                                        insertLogInfo(username, "Api Login Successfull");
                                                    else
                                                        insertLogInfo(username, "Api Error Login Not Successfull :" + logInResult.Result.Info.Message.ToString());
### HERE I KNOW IT IS SHOWING THE CHALLENGE HOW DO I GET THE URL PROPERLY ITS NOT IN TEH ABOVE "logInResult "

                                                }

Thank you

**Comment by vitalragaz** [comment]
@EezeSolutions 
First of all I'm not sure if you're working on the right layer of the code. As I see from your code you try to solve that issue over the given endpoints of the api instead of modifying the core logic. 
check my  [Screenshot i've posted a while ago](https://user-images.githubusercontent.com/3964460/33828394-3dd57200-de6c-11e7-9a94-23cdc3cf850f.png). The correct position of the change would be there. The json object that i've created for demo purpose should show you how you can get the url.

Cheers

**Comment by EezeSolutions** [comment]
@vitalragaz   Done  :) thank you for guiding to the correct path. I have the dll now with additional login function which will send back the URL .

And another function that will than use that URL to send code to phone / email and when we put it back the check is passed woohooo :dancing_men:
